src/dashboard/src/installer/middleware.py

```diff
@@ -30,11 +30,8 @@
 
 class ConfigurationCheckMiddleware:
     def process_request(self, request):
-        if User.objects.count() == 0:
+        # The presence of the UUID is an indicator of whether we've already set up.
```

Comment:
The first user is already auto generated by shibboleth by the time we get here, so we have to use something else to decide whether to display the welcome screen.